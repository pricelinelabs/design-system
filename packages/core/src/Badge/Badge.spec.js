import React from 'react'
import { Badge, theme } from '..'

describe('Badge', () => {
  let consoleError
  beforeEach(() => {
    consoleError = console.error
    console.error = jest.fn()
  })
  afterEach(() => (console.error = consoleError))

  test('renders', () => {
    const json = rendererCreateWithTheme(<Badge />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders small', () => {
    const json = rendererCreateWithTheme(<Badge size='small' />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('color secondary sets background-color and color', () => {
    const json = rendererCreateWithTheme(<Badge color='secondary' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.green)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('color secondary.light sets background-color and color', () => {
    const json = rendererCreateWithTheme(
      <Badge color='secondary.light' />
    ).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGreen)
    expect(json).toHaveStyleRule('color', theme.colors.text)
  })

  test('color error sets background-color and color', () => {
    const json = rendererCreateWithTheme(<Badge color='error' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.red)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('color error.light sets background-color and color', () => {
    const json = rendererCreateWithTheme(<Badge color='error.light' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightRed)
    expect(json).toHaveStyleRule('color', theme.colors.text)
  })

  test('color alert sets background-color and color', () => {
    const json = rendererCreateWithTheme(<Badge color='alert' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.orange)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('color primary sets background-color and color', () => {
    const json = rendererCreateWithTheme(<Badge color='primary' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.blue)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('color primary.light sets background-color and color', () => {
    const json = rendererCreateWithTheme(
      <Badge color='primary.light' />
    ).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightBlue)
    expect(json).toHaveStyleRule('color', theme.colors.text)
  })
})
