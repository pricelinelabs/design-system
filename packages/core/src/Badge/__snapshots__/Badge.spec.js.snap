// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge color alert sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #f68013;
  color: #fff;
}

<div
  className="c0"
  color="alert"
  size="medium"
/>
`;

exports[`Badge color error sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #c00;
  color: #fff;
}

<div
  className="c0"
  color="error"
  size="medium"
/>
`;

exports[`Badge color error.light sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #fbebeb;
  color: #001833;
}

<div
  className="c0"
  color="error.light"
  size="medium"
/>
`;

exports[`Badge color primary sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #0068EF;
  color: #fff;
}

<div
  className="c0"
  color="primary"
  size="medium"
/>
`;

exports[`Badge color primary.light sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #e8f2ff;
  color: #001833;
}

<div
  className="c0"
  color="primary.light"
  size="medium"
/>
`;

exports[`Badge color secondary sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #0a0;
  color: #fff;
}

<div
  className="c0"
  color="secondary"
  size="medium"
/>
`;

exports[`Badge color secondary.light sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #ecf7ec;
  color: #001833;
}

<div
  className="c0"
  color="secondary.light"
  size="medium"
/>
`;

exports[`Badge renders 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  color: #001833;
  background-color: #f4f6f8;
}

<div
  className="c0"
  color="text"
  size="medium"
/>
`;

exports[`Badge renders small 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 10px;
  font-weight: 700;
  line-height: 1.4;
  line-height: 1.25;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  color: #001833;
  background-color: #f4f6f8;
}

<div
  className="c0"
  color="text"
  size="small"
/>
`;
